5.2.Cup_of_join.py

```diff
@@ -0,0 +1,16 @@
+def join(*lists,sep='-'):
+    l=[]
+    for i in range(len(lists)):
+        for j in lists[i]:
+            l.append(j)
+        if (i + 1 != len(lists)):
+            l.append(sep)
+    if not(l):
```

Comment:
1. rename 'i' 'j' and 'l'

2. Try to find a way to make the code cleaner, 
Replace these two for loops with a list comprehention with two iterators - 
https://stackoverflow.com/questions/18551458/how-to-frame-two-for-loops-in-list-comprehension-python

Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features